### PR TITLE
Added a new doc for covert from k8s deploy to knative service

### PR DESCRIPTION
**Pull request by gyliu513**
Fixes https://github.com/knative/docs/issues/3841

/cc @evankanderson 

## Proposed Changes 

-
-
-

### CODE REVIEW

File: docs/serving/convert-deployment-to-knative-service.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Covert Kubernetes Deployment to Knative Service
+
+Learn how to covert Kubernetes Deployment to Knative Service.
```
Comment (snneji) [inline docs/serving/convert-deployment-to-knative-service.md:3]:
```suggestion
This topic shows how to convert a Kubernetes Deployment to a Knative Service.
```

File: docs/serving/convert-deployment-to-knative-service.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+## Determining if your workload is a good fit for Knative
```
Comment (snneji) [inline docs/serving/convert-deployment-to-knative-service.md:5]:
```suggestion
## Determine if your workload is a good fit for Knative
```

File: docs/serving/convert-deployment-to-knative-service.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+In general, if your Kubernetes workload is a good fit for Knative, you should be able to remove a lot of your manifest to create a Knative Service.
+
+There are three aspects you need to consider:
+
+- All work done is triggered by HTTP
+- The container is stateless (all state is stored elsewhere or can be re-created)
```
Comment (snneji) [inline docs/serving/convert-deployment-to-knative-service.md:12]:
```suggestion
- The container is stateless. All state is stored elsewhere or can be re-created.
```

File: docs/serving/convert-deployment-to-knative-service.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+Learn how to covert Kubernetes Deployment to Knative Service.
+
+## Determining if your workload is a good fit for Knative
+
+In general, if your Kubernetes workload is a good fit for Knative, you should be able to remove a lot of your manifest to create a Knative Service.
+
+There are three aspects you need to consider:
+
+- All work done is triggered by HTTP
```
Comment (snneji) [inline docs/serving/convert-deployment-to-knative-service.md:11]:
```suggestion
- All work done is triggered by HTTP.
```

File: docs/serving/convert-deployment-to-knative-service.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+- The container is stateless (all state is stored elsewhere or can be re-created)
+- Uses only Secret and ConfigMap volumes
+
+## Example conversion
+
+The following will use an Kubernetes Nginx Deployment and Service, and covert it to Knative Service.
```
Comment (snneji) [inline docs/serving/convert-deployment-to-knative-service.md:17]:
```suggestion
The following example uses a Kubernetes Nginx Deployment and Service, and converts it to a Knative Service.
```

**Review by snneji — CHANGES_REQUESTED** [review]

File: docs/serving/convert-deployment-to-knative-service.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
+# Covert Kubernetes Deployment to Knative Service
+
+Learn how to covert Kubernetes Deployment to Knative Service.
+
+## Determining if your workload is a good fit for Knative
+
+In general, if your Kubernetes workload is a good fit for Knative, you should be able to remove a lot of your manifest to create a Knative Service.
```
Comment (snneji) [inline docs/serving/convert-deployment-to-knative-service.md:7]:
```suggestion
In general, if your Kubernetes workload is a good fit for Knative, you can remove a lot of your manifest to create a Knative Service.
```

File: docs/serving/convert-deployment-to-knative-service.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
+# Covert Kubernetes Deployment to Knative Service
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md]: Done

File: docs/serving/convert-deployment-to-knative-service.md (hunk 1 above)
Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md]: Done

File: docs/serving/convert-deployment-to-knative-service.md (hunk 2 above)
Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md]: Done

File: docs/serving/convert-deployment-to-knative-service.md (hunk 6 above)
Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md]: Done

File: docs/serving/convert-deployment-to-knative-service.md (hunk 4 above)
Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md]: Done

File: docs/serving/convert-deployment-to-knative-service.md (hunk 3 above)
Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md]: Done

File: docs/serving/convert-deployment-to-knative-service.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
+## Determining if your workload is a good fit for Knative
+
+In general, if your Kubernetes workload is a good fit for Knative, you should be able to remove a lot of your manifest to create a Knative Service.
+
+There are three aspects you need to consider:
+
+- All work done is triggered by HTTP
+- The container is stateless (all state is stored elsewhere or can be re-created)
+- Uses only Secret and ConfigMap volumes
```
[the remark this answers is not in the document]
Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md]: Done

File: docs/serving/convert-deployment-to-knative-service.md (hunk 5 above)
Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md]: Done

**Review by gyliu513 — COMMENTED** [review]
Thanks @snneji , all comments are addressed.

File: docs/serving/convert-deployment-to-knative-service.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+There are three aspects you need to consider:
+
+- All work done is triggered by HTTP.
+- The container is stateless. All state is stored elsewhere or can be re-created.
+- Your workload uses only Secret and ConfigMap volumes.
+
+## Example conversion
+
+The following example shows a Kubernetes Nginx Deployment and Service, and shows how it converts to a Knative Service.
```
Comment (evankanderson) [inline docs/serving/convert-deployment-to-knative-service.md:17]: Reference the k8s docs this was taken from?

**Review by evankanderson — COMMENTED** [review]
/approve

File: docs/serving/convert-deployment-to-knative-service.md (hunk 9 above)
Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md]: @evankanderson Done, thanks!

**Comment by omerbensaadon** [comment]
This looks great! I'm going to add it to the Navigation via mkdocs.yml and also link to it from the end of Getting Started :)

**Comment by omerbensaadon** [comment]
@abrennan89 @snneji @RichardJJG feel free to move this to a different location in a follow-on PR if I've got this wrong :D

**Comment by omerbensaadon** [comment]
@googlebot I consent.

**Comment by abrennan89** [comment]
@omerbensaadon I think this should be in the developer guide probably

File: docs/serving/convert-deployment-to-knative-service.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+
+## Determine if your workload is a good fit for Knative
+
+In general, if your Kubernetes workload is a good fit for Knative, you can remove a lot of your manifest to create a Knative Service.
+
+There are three aspects you need to consider:
+
+- All work done is triggered by HTTP.
+- The container is stateless. All state is stored elsewhere or can be re-created.
+- Your workload uses only Secret and ConfigMap volumes.
```
Comment (markusthoemmes) [inline docs/serving/convert-deployment-to-knative-service.md:13]: I think this should also include autoscaling. The shown transformation would include scale the nginx pod for roughly every 100 concurrent HTTP requests, so you're saving even more YAML :)

Comment (gyliu513) [reply docs/serving/convert-deployment-to-knative-service.md:13]: Good comment, thanks @markusthoemmes .

How about add a new section named as follow:

## Benefits

- Reduce footprint of the service instance as the instance will be scale to 0 when it became idle.
- Performance can be improved via build-in autoscaling for knative service.

Thoughts?

**Pull request merged**